Orange/widgets/visualize/owradviz.py

```diff
@@ -360,19 +346,18 @@ def set_data(self, data):
         self._vizrank_color_change()
 
     def check_data(self):
+        def error(err):
+            err()
+            self.data = None
+
         super().check_data()
         if self.data is not None:
             domain = self.data.domain
             if len(self.data) < 2:
-                self.Error.no_instances()
-                self.data = None
+                error(self.Error.no_instances)
             elif len([v for v in domain.variables + domain.metas
                       if v.is_primitive()]) < 3:
-                self.Error.no_features()
-                self.data = None
-            elif np.all(np.isnan(self.data.X).any(axis=1)):
-                self.Warning.missing_coords()
-                self.data = None
+                error(self.Error.no_features)
 
     def init_attr_values(self):
         super().init_attr_values()
```

Comment:
Add

```
def primitive_variables(self):
    return [v for v in chain(domain.attributes, domain.metas)
                         if v.is_primitive()]
```

and call it from `set_data`, `check_data`, `init_attr_values` instead of doing this same in all of them. If `len(self.model_other) + len(self.model_selected) > 3` in `_vizrank_color_changed` is still needed, it can be replaced with checking `len(self.primitive_variables)`, too.